src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -319,6 +344,12 @@ uct_cuda_ipc_md_open(uct_component_t *component, const char *md_name,
         return UCS_ERR_NO_MEMORY;
     }
 
+    /* assumes device oridnal 0 is available and initialized */
+    if (CUDA_SUCCESS != cuDeviceTotalMem(&total_bytes, 0)) {
+        return UCS_ERR_IO_ERROR;
+    }
```

Comment:
@rakhmets We don't need this check here because we are trying to set rcache limits when the remote caching gets used (either during rkey_unpack or during an actual transfer), at which point if there is no context we should fail. Given that cuDeviceTotalMem doesn't require a context to be active it should be safe to leave logic here unchanged. WDYT?

Reply:
I think it is more precisely to query information for current device, instead device 0. If it is assumed, that the context should be set, then it is better to track the opposite case.

```
if (!uct_cuda_base_is_context_active()) {
    ucs_error("no active context");
    return UCS_ERR_IO_ERROR;
}

if (UCT_CUDADRV_FUNC_LOG_DEBUG(cuCtxGetDevice(&device)) != UCS_OK) {
    return UCS_ERR_IO_ERROR;
}